topicExtractor.py

```diff
@@ -0,0 +1,169 @@
+from config import OpenAIBot, LangChainBot
+from utils import dataLoader, dataSaver, printAndLog
+from bertopic import BERTopic
+from keybert import KeyBERT
+from sklearn.feature_extraction.text import CountVectorizer
+from langchain_community.callbacks import get_openai_callback
+
+
+def retrieveTopics(
+    videoToUse, transcriptToUse, config, overwrite=False, saveNameAppend=""
+):
+    """
+    Retrieves or generates topic model and data.
+
+    Args:
+        videoToUse (str): Path to the video file.
+        transcriptToUse (str): Path to the transcript file.
+        config (dict): Configuration settings.
+        overwrite (bool, optional): Whether to overwrite existing data. Defaults to False.
+        saveNameAppend (str, optional): String to append to the save name. Defaults to "".
+
+    Returns:
+        tuple: A tuple containing the topics over time and the topic model.
+    """
+
+    if not overwrite and (
+        (topicModel := dataLoader(config, "topicModel", videoToUse, saveNameAppend))
+        is not False
+        and (
+            topicsOverTime := dataLoader(
+                config, "topicsOverTime", videoToUse, saveNameAppend
+            )
+        )
+        is not False
+    ):
```

Comment:
Optional: the block of code from L27-33 is not reading friendly. 

I would extract `dataLoader(config, "topicModel", videoToUse, saveNameAppend))` and `dataLoader(
                config, "topicsOverTime", videoToUse, saveNameAppend
            )` as separate variable and make the `if` call simple to read.

Reply:
Implemented in the new [Commit](https://github.com/tl-its-umich-edu/annoto-gai/pull/14/commits/12992ec3fadf5a4771519490c1f580d4f442db7b). 
I was trying the new Walrus operator in Python out and liked the way it worked. But for readability, not so much.

Reply:
>trying the new Walrus operator in Python out

I like the use of it. But I felt code readability is important. This has 3 `and` operator and I had halt and focus and see what's going in the code.